airflow/www/static/js/grid/details/content/taskInstance/Logs/index.jsx

```diff
@@ -0,0 +1,191 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useRef, useState, useEffect } from 'react';
+import {
+  Text,
+  Box,
+  Flex,
+  Divider,
+  Code,
+  Button,
+  Checkbox,
+} from '@chakra-ui/react';
+
+import { getMetaValue } from '../../../../../utils';
+import LogLink from './LogLink';
+import useTaskLog from '../../../../api/useTaskLog';
+import LinkButton from '../../../../components/LinkButton';
+
+const showExternalLogRedirect = getMetaValue('show_external_log_redirect') === 'True';
+const externalLogName = getMetaValue('external_log_name');
+const logUrl = getMetaValue('log_url');
+
+const getLinkIndexes = (tryNumber) => {
+  const internalIndexes = [];
+  const externalIndexes = [];
+
+  [...Array(tryNumber + 1 || 0)].forEach((_, index) => {
+    if (index === 0 && tryNumber < 2) return;
+    const isExternal = index !== 0 && showExternalLogRedirect;
+    if (isExternal) {
+      externalIndexes.push(index);
+    } else {
+      internalIndexes.push(index);
+    }
+  });
+
+  return [internalIndexes, externalIndexes];
+};
+
+const Logs = ({
+  dagId,
+  dagRunId,
+  taskId,
+  executionDate,
+  tryNumber,
+  isGroup,
+}) => {
+  const [internalIndexes, externalIndexes] = getLinkIndexes(tryNumber);
+  const [selectedAttempt, setSelectedAttempt] = useState(1);
+  const [shouldRequestFullContent, setShouldRequestFullContent] = useState(false);
+  const [wrap, setWrap] = useState(false);
+  const { data, isSuccess } = useTaskLog({
+    dagId,
+    dagRunId,
+    taskId,
+    taskTryNumber: selectedAttempt,
+    fullContent: shouldRequestFullContent,
+    enabled: (!isGroup),
+  });
+
+  const codeBlockBottomDiv = useRef(null);
+
+  useEffect(() => {
+    if (codeBlockBottomDiv.current) {
+      codeBlockBottomDiv.current.scrollIntoView({ behavior: 'smooth' });
```

Comment:
Actually, I think it may be a better UX if we don't show any animation here. And probably show a loading indicator until the bottom scroll is complete.

Reply:
Also, let's make sure we scroll to the bottom when toggling wrap/no-wrap.

Reply:
Good idea.
1 - Done for the animation.  For the indicator during scroll, I am not sure how to detect when the scrollIntoView is done. We could use `IntersectionObserver` for instance but it feels kind of over kill for this particular use case. Do you see a simple way for achieving that ?

2 - Done

Reply:
What you have now is good.